rules/rule_manager.go

```diff
@@ -1,22 +1,27 @@
 package rules
 
 import (
+	"maps"
+	"slices"
+	"sort"
 	"strings"
 )
 
 type ruleManager struct {
 	constraints        map[string]constraint
 	currentIndex       int
 	rulesBySlashCount  map[int]map[DynamicRule]int
-	prefixes           map[string]string
+	prefixes           map[string]uint
+	watcherPrefixes    map[string]uint
 	rules              []DynamicRule
 	enhancedRuleFilter bool
 }
 
 func newRuleManager(constraints map[string]constraint, enhancedRuleFilter bool) ruleManager {
 	rm := ruleManager{
 		rulesBySlashCount:  map[int]map[DynamicRule]int{},
-		prefixes:           map[string]string{},
+		prefixes:           map[string]uint{},
+		watcherPrefixes:    map[string]uint{},
```

Comment:
Do we really need to store these here?  Could we instead use a 
`prefixes: map[string]struct{crawlerOnly: bool, priority: uint}` and can generate any combination from that?